python/ray/ml/train/integrations/rl/rl_trainer.py

```diff
@@ -0,0 +1,201 @@
+            )
+
+            trainer = RLTrainer(
+                run_config=RunConfig(stop={"training_iteration": 5}),
+                scaling_config={
+                    "num_workers": 2,
+                    "use_gpu": False,
+                },
+                datasets={"train": dataset},
+                algorithm=BCTrainer,
+                config={
+                    "env": "CartPole-v0",
+                    "framework": "tf",
+                    "evaluation_num_workers": 1,
+                    "evaluation_interval": 1,
+                    "evaluation_config": {"input": "sampler"},
+                },
+            )
+            result = trainer.fit()
+
+    """
+
+    def __init__(
+        self,
+        algorithm: Union[str, Type[RLLibTrainer]],
+        config: Optional[Dict[str, Any]] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
+        preprocessor: Optional[Preprocessor] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+        **train_kwargs
+    ):
+        self._algorithm = algorithm
+        self._train_kwargs = train_kwargs
+        self._config = config if config is not None else {}
+
+        Trainer.__init__(
+            scaling_config=scaling_config,
+            run_config=run_config,
+            datasets=datasets,
+            preprocessor=preprocessor,
+            resume_from_checkpoint=resume_from_checkpoint,
+        )
+
+    def _get_rllib_config(self, process_datasets: bool = False) -> Dict:
+        config = self._config.copy()
+        num_workers = self.scaling_config.get("num_workers")
+        if num_workers is not None:
+            config["num_workers"] = num_workers
+
+        worker_resources = self.scaling_config.get("resources_per_worker")
+        if worker_resources:
+            res = worker_resources.copy()
+            config["num_cpus_per_worker"] = res.pop("CPU", 1)
+            config["num_gpus_per_worker"] = res.pop("GPU", 0)
+            config["custom_resources_per_worker"] = res
+
+        use_gpu = self.scaling_config.get("use_gpu")
+        if use_gpu:
+            config["num_gpus"] = 1
+
+        trainer_resources = self.scaling_config.get("trainer_resources")
+        if trainer_resources:
+            config["num_cpus_for_driver"] = trainer_resources.get("CPU", 1)
+
+        if process_datasets:
+            self.preprocess_datasets()
+            # Up for discussion: If datasets is passed, should we always
+            # set the input config? Is the sampler config required here, too?
+            if self.datasets:
+                config["input"] = "dataset"
+                config["input_config"] = {
+                    "loader_fn": lambda: self.datasets["train"],
+                }
+
+        return config
+
+    def training_loop(self) -> None:
```

Comment:
This makes me think that we should reconsider our interface for `Trainer`. Currently we make the assumption that `training_loop` will be overridden by all subclasses (hence why it is an abstract method), but is not the case here.

Reply:
(this is just a thought- it should not block this PR!)